shared/triage_metadata.go

```diff
@@ -0,0 +1,257 @@
+	if err != nil {
+		return err
+	}
+
+	tm.logger.Infof("PR created: %s", pr.GetHTMLURL())
+	return nil
+}
+
+func (tm triageMetadata) createWPTMetadataPR(triagedMetadataMap map[string][]byte) error {
+	log := tm.logger
+	ref, err := tm.getCommitBranchRef()
+	if err != nil {
+		log.Errorf("Unable to get/create the commit reference: %s", err)
+		return err
+	}
+
+	if ref == nil {
+		log.Errorf("No error returned but the reference is nil")
+		return errors.New("No error returned but the reference is nil")
+	}
+
+	tree, err := tm.getTree(ref, triagedMetadataMap)
+	if err != nil {
+		log.Errorf("Unable to create the tree based on the provided files: %s", err)
+		return err
+	}
+
+	if err := tm.pushCommit(ref, tree); err != nil {
+		log.Errorf("Unable to create the commit: %s", err)
+		return err
+	}
+
+	if err := tm.createPR(); err != nil {
+		log.Errorf("Error while creating the pull request: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+// Add Metadata into the existing Metadata YML files and only return modified files.
+func addToFiles(metadata MetadataResults, filesMap map[string]Metadata, logger Logger) map[string][]byte {
+	// Update filesMap with the new information in metadata.
+	for test, links := range metadata {
+		folderName, _ := SplitWPTTestPath(test)
+		appendTestName(test, metadata)
+		// If the META.YML does not exist in the repository.
+		if _, ok := filesMap[folderName]; !ok {
+			filesMap[folderName] = Metadata{Links: links}
+			continue
+		}
+
+		// Folder already exists.
+		for _, link := range links {
+			existingMetadata := filesMap[folderName]
+			hasMerged := false
+			for index, existingLink := range existingMetadata.Links {
+				if link.URL == existingLink.URL && link.Product.MatchesProductSpec(existingLink.Product) {
+					// Add new MetadataResult to the existing link.
+					filesMap[folderName].Links[index].Results = append(existingMetadata.Links[index].Results, link.Results...)
+					hasMerged = true
+					break
+				}
+			}
+
+			// Add new MetadataLink to the existing Link if no link was found.
+			if !hasMerged {
+				filesMap[folderName] = Metadata{Links: append(filesMap[folderName].Links, link)}
+			}
+		}
+	}
+
+	// Grab all newly updated metadata files.
+	res := make(map[string][]byte)
+	for test := range metadata {
+		folderName, _ := SplitWPTTestPath(test)
+		metadataBytes, err := yaml.Marshal(filesMap[folderName])
+		if err != nil {
+			logger.Errorf("Error from marshal %s: %s", folderName, err.Error())
+			continue
+		}
+		res[folderName] = metadataBytes
+	}
+	return res
+}
+
+// appendTestName populate TestPath name of metadata from test.
+func appendTestName(test string, metadata MetadataResults) {
+	links := metadata[test]
+	_, testName := SplitWPTTestPath(test)
+	for linkIndex, link := range links {
+		for resultIndex := range link.Results {
+			metadata[test][linkIndex].Results[resultIndex].TestPath = testName
+		}
+	}
+}
+
+func generateRandomInt() string {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
No need to seed `rand` every time you use it; this can be in an init method.

Also note that this is not cryptographically secure of course, but that isn't important here.